jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpChannelOverHTTP2.java

```diff
@@ -655,8 +655,10 @@ public boolean onTimeout(Throwable failure, Consumer<Runnable> consumer)
         getHttpTransport().onStreamTimeout(failure);
 
         failure.addSuppressed(new Throwable("HttpInput idle timeout"));
-        _contentDemander.onTimeout(failure);
-        boolean needed = getRequest().getHttpInput().onContentProducible();
+        boolean readFailed = _contentDemander.onTimeout(failure);
+        boolean needed = false;
+        if (readFailed)
```

Comment:
I think a simple && would be clear here.
Needed = on timeout && contentProducable

Reply:
@gregw I disagree that it would be clearer to lose the variable names and rely on `&&` short circuiting.
I have renamed `needed` to `handle`.

Reply:
Nah! You can keep the variable name but get rid of the manual short-circuit:

boolean = readFailed && getRequest().getHttpInput().onContentProducible();